moto/elbv2/models.py

```diff
@@ -214,30 +215,23 @@ def __init__(
         self.certificate = certificate
         self.certificates = [certificate] if certificate is not None else []
         self.default_actions = default_actions
-        self._non_default_rules = []
-        self._default_rule = FakeRule(
-            listener_arn=self.arn,
-            conditions=[],
-            priority="default",
-            actions=default_actions,
-            is_default=True,
-        )
+        self._non_default_rules = OrderedDict()
 
     @property
     def physical_resource_id(self):
         return self.arn
 
     @property
     def rules(self):
-        return self._non_default_rules + [self._default_rule]
+        return self._non_default_rules
 
-    def remove_rule(self, rule):
-        self._non_default_rules.remove(rule)
+    def remove_rule(self, arn):
+        self._non_default_rules.pop(arn)
 
-    def register(self, rule):
-        self._non_default_rules.append(rule)
-        self._non_default_rules = sorted(
-            self._non_default_rules, key=lambda x: x.priority
+    def register(self, arn, rule):
+        self._non_default_rules[arn] = rule
+        self._non_default_rules = OrderedDict(
+            reversed(list(self._non_default_rules.items()))
```

Comment:
What's the reasoning behind this change?

Reply:
I updated it to be an `OrderedDict` instead of a `list` like other "FakeServices". If I do not reverse it then the order of the items inserted is changed. The order in which the items are tested against will changer here:
https://github.com/spulec/moto/blob/master/tests/test_elbv2/test_elbv2.py#L1113-L1115 i.e. `[0]` element will become he `[2]` element ec.

Reply:
Oh yes. The `Action` works a little differently in `ListernRule`. My intention was to fix that in the next PR. 
Here is how: There is a [`ForwardConfig` option in the `ListenerRule`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-elasticloadbalancingv2-listenerrule-action.html#cfn-elasticloadbalancingv2-listenerrule-action-forwardconfig) that in-turn defines the `TargetGroups` and the `TargetGroupStickinessConfig`. That needs to be added in the check. So really this will need to change when adding proper validations.
What do you think?

Reply:
Example:
```
LoadBalancerListenerRule:
    Type: AWS::ElasticLoadBalancingV2::ListenerRule
    Properties:
      Actions:
        - Type: forward
          ForwardConfig:
            TargetGroups:
              - TargetGroupArn:
                  {{target_group.arn}}
                Weight: {{target_group.weight}}
            TargetGroupStickinessConfig:
              DurationSeconds: 300
              Enabled:
                Fn::If: [ IsStickinessEnabled, true, false ]
      Conditions:
          - Field: path-pattern
            Values:
              - '/*'
      ListenerArn: 
        Ref: LoadBalancerListener
      Priority: 1
```

This, right now, fails with the error:
KeyError: 'TargetGroupArn'

Reply:
Sorry, I wasn't very clear. Why the change from sorting it by priority?
I don't have any real world experience with these methods, so not sure what AWS does. But from the docs I can see that the CLI does show them in priority order

Reply:
Fixed: eb95069
Also changed the priority order from 50 to 500 to show that lowest priority comes first